app/src/services/ApiService.js

```diff
@@ -16,38 +18,74 @@ export default class ApiService {
     };
     //load env provided URL at query time to allow conf.js to load it in time
     //in testing
-    const ALL_SERVICE_URL = window.envConfig.KATLAS_API_URL;
-    return requestHelper(ALL_SERVICE_URL + ALL_SERVICE_CONTEXT +
-      QUERY_KEYWORD_SERVICE_PATH, params);
+    return requestHelper(
+      getServiceURL() + ALL_SERVICE_CONTEXT + QUERY_KEYWORD_SERVICE_PATH,
+      params
+    );
   }
 
-  static getQSLResult(query) {
-    const params = {
-      [QUERY_QSL_PARAM_NAME]: query
-    };
+  static getQSLResult(query, page, rowsPerPage) {
     //load env provided URL at query time to allow conf.js to load it in time
     //in testing
-    const ALL_SERVICE_URL = window.envConfig.KATLAS_API_URL;
-    return requestHelper(ALL_SERVICE_URL + ALL_SERVICE_CONTEXT +
-      QUERY_QSL_SERVICE_PATH, params);
+    const querySegments = query.split('.');
+    const rootEntityQuery = querySegments[0];
+    // inject the pagination if not provided
+    if (!rootEntityQuery.includes('$$')) {
+      const matches = /([a-zA-Z0-9]+)\[?(?:(@[",@$=><!a-zA-Z0-9\-.|&:_]*|\**|\$\$[a-zA-Z0-9,=]+))\]?\{([*|[,@"=a-zA-Z0-9-]*)/.exec(
+        rootEntityQuery
+      );
+      if (matches) {
+        const objType = matches[1];
+        const filter = matches[2];
+        const fields = matches[3];
+        const pagination = `$$first=${rowsPerPage},offset=${page*rowsPerPage}`
+        querySegments[0] = `${objType}[${filter}${pagination}]{${fields}}`
+      }
+    }
+
+    return requestHelper(
+      getServiceURL() +
+        ALL_SERVICE_CONTEXT +
+        QUERY_QSL_SERVICE_PATH +
+        querySegments.join('.')
+    );
   }
 
   static getEntity(uid) {
     //load env provided URL at query time to allow conf.js to load it in time
     //in testing
-    const ALL_SERVICE_URL = window.envConfig.KATLAS_API_URL;
-    return requestHelper(ALL_SERVICE_URL + ALL_SERVICE_CONTEXT +
-      ENTITY_SERVICE_PATH + uid);
+    return requestHelper(
+      getServiceURL() + ALL_SERVICE_CONTEXT + ENTITY_SERVICE_PATH + uid
+    );
   }
 }
 
+const getServiceURL = () => {
+  return process.env.NODE_ENV === 'test'
```

Comment:
@fzxu, might it be easier simply for the `KATLAS_API_URL` value to equal localhost when run outside of a deployed env?